zerver/tests/test_urls.py

```diff
@@ -128,9 +128,10 @@ def test_config_error_endpoints_dev_env(self) -> None:
 
 
 class URLResolutionTest(ZulipTestCase):
-    def get_callback_string(self, pattern: django.urls.resolvers.URLPattern) -> Optional[str]:
-        callback_str = "lookup_str" if hasattr(pattern, "lookup_str") else "_callback_str"
-        return getattr(pattern, callback_str, None)
+    def get_callback_string(
+        self, pattern: Union[django.urls.resolvers.URLPattern, django.urls.resolvers.URLResolver]
+    ) -> Optional[str]:
+        return pattern.lookup_str if isinstance(pattern, django.urls.resolvers.URLPattern) else None
```

Comment:
This function is not a useful abstraction anymore. Its only caller would be better written directly:

```diff
     def test_non_api_url_resolution(self) -> None:
         for pattern in urls.urlpatterns:
-            callback_str = self.get_callback_string(pattern)
-            if callback_str:
-                (module_name, base_view) = callback_str.rsplit(".", 1)
+            if isinstance(pattern, URLPattern):
+                (module_name, base_view) = pattern.lookup_str.rsplit(".", 1)
                 self.check_function_exists(module_name, base_view)
```

Reply:
OK, fixed.